### PR TITLE
[ServiceBus/EventHub] move async ssl opts to transport constructor

### PR DESCRIPTION
**Pull request by swathipil**
addresses: #37246

Creating the SSLContext and loading verify locations [can be done in the constructor.](https://github.com/python/cpython/blob/7e7223e18f58ec48fb36a68fb75b5c5b7a45042a/Lib/asyncio/sslproto.py#L55) Moving the check to the constructor so it doesn't need to be run in a ThreadPoolExecutor.

TODO:
- [x] test/confirm
   - [x] test_sb_client_async.py:test_custom_endpoint_connection_verify_exception_async
   - [x] test_negative_async.py:test_client_invalid_credential_async

### CODE REVIEW

File: sdk/servicebus/azure-servicebus/tests/test_sb_client.py [hunk 1]
```diff
@@ -700,8 +700,9 @@ def test_custom_endpoint_connection_verify_exception(self,
         # invalid cert file to connection_verify should fail
         client = ServiceBusClient(hostname, credential, connection_verify="fakecertfile.pem", uamqp_transport=uamqp_transport)
         with client:
+            sender = client.get_queue_sender(servicebus_queue.name)
             with pytest.raises(ServiceBusError):
-                with client.get_queue_sender(servicebus_queue.name) as sender:
```
Comment (kashifkhan) [inline sdk/servicebus/azure-servicebus/tests/test_sb_client.py:704]: out of curiosity, why was this change needed?

Comment (swathipil) [reply sdk/servicebus/azure-servicebus/tests/test_sb_client.py:704]: I just wanted to make sure that the error was happening in the same place (when we enter with the context manager), rather than when the object is created. So I separated this into two statements.

**Review by l0lawrence — APPROVED** [review]
lgtm :)

**Review by kashifkhan — APPROVED** [review]

**Comment by sveinse** [comment]
Is it an acceptable solution to move blocking IO to the constructor of the class? A common use case of these objects is to dynamically create `ServiceBusClient` instances from coroutines which is under a running async event loop. Any blocking IO during initialization will restrict when and where in an async app these objects can be created, which I believe is limiting the usefulness of the fix. Thank you.

(I'm the reporter for the issue #37246)

**Comment by swathipil** [comment]
Hi @sveinse - We will discuss and let you know. Thanks.

**Comment by sveinse** [comment]
@swathipil thank you. If you're not aware of the comments in #37246 , please consider the new comments https://github.com/Azure/azure-sdk-for-python/issues/37246#issuecomment-2387058474 and onwards. Thanks.
